platform/event/gossiped_event.proto

```diff
@@ -0,0 +1,61 @@
+/**
+ * # Gossiped Event
+ * An event that is sent and received via gossip
+ *
+ * ### Keywords
+ * The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+ * "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+ * document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119).
+ */
+syntax = "proto3";
+
+package com.hedera.hapi.platform.event;
+
+/*
+ * Hedera Network Services Protobuf
+ *
+ * Copyright (C) 2018 - 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import "event_payload.proto";
+import "core_event_data.proto";
+
+option java_package = "com.hedera.hapi.platform.event";
+// <<<pbj.java_package = "com.hedera.hapi.platform.event">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * An event that is sent and received via gossip
+ */
+message GossipedEvent {
+  /**
+   * The core event data
+   */
+  CoreEventData core_event_data = 1;
+
+  /**
+   * The signature of the event creator on the hash of the event. <br/>
+   * The signed hash MUST have been created using the same methodology as the hash in an EventDescriptor. <br/>
+   * The signature MUST have been created with the private key corresponding to the public key belonging to the event
+   * creator, as it exists in the current address book.
+   */
+  bytes signature = 2;
```

Comment:
We need to define what kind of signature this is, RSA, HinTS etc, similar problem to Hashes commented above.

Reply:
Adding a `future work` comment to specify signature type once the decision has been made

Reply:
currently we use SHA384withRSA

Reply:
Any thought to perhaps using edDSA instead of RSA at some point?  RSA signatures are much larger (and hence cost much more bandwidth) than edDSA.

Reply:
I went ahead and specified `SHA384withRSA` in the doc, since it is the status quo, and none of the alternatives are a clear improvement:

- a BLS signature is small but extremely expensive to verify, and not quantum secure
- ed25519 is moderately smaller than RSA, but also not quantum secure

Reply:
In fairness, RSA isn't quantum secure either.  I don't know of any public key algorithm currently that's quantum secure (which might be a problem, except practical quantum computing is still quite a ways off, in my opinion).

Reply:
👍 My main argument for RSA is just status quo.

Reply:
Understood.
I wasn't arguing for an immediate change.
The signature size has a significant impact on block stream data volume, however, so it might be worth _considering_ a future change.